### PR TITLE
fix: message bar icon in high contrast, and restyle message bar button/link

### PR DESCRIPTION
**Pull request by patrickhlauke**
#### Details

* set icon to use `$always-black` rather than `$secondary-text`, so it doesn't end up white on light yellow in high contrast mode
* restyle button/link in message bar to always use underline and to inherit the regular text color of the message bar - avoids yellow button/link on light yellow in high contrast, but also improves visibility of the link in regular contrast mode

##### Motivation

Closes https://github.com/microsoft/accessibility-insights-web/issues/5757

In regular contrast mode, any button/link inside the message bar is not very obvious - and the difference between the blue button/link and the adjacent static text is relatively low.

More importantly/critically, when set to high contrast, the little "i" info icon turns white on light yellow, and the button/link uses yellow text on light yellow making both completely illegible.

This PR attempts to fix these issues with minimal styling changes.

---

###### Screenshots

**Before:**

Regular contrast, icon and button/link (dark gray/nearly black icon and blue button/link on light yellow bar)

Regular contrast, icon and button/link

High contrast, icon and button/link (white icon and yellow button/link on light yellow message bar)

High contrast, icon and button/link

**After:**

Regular contrast, icon and button/link (black icon and dark gray/black button/link with underline on light yellow bar)

Regular contrast, icon and button/link

High contrast, icon and button/link (black icon and dark gray/black button/link with underline on light yellow bar)

High contrast, icon and button/link

##### Context

Admittedly the change to using dark text with underline for the button/link may be a bit opinionated, and a slight departure from links anywhere else in the extension. I'd argue though that even in those other cases, it's worth considering making at least the underline visible for the links so as not to just rely mostly on colour to distinguish them from static text.

#### Pull request checklist

- [x] Addresses an existing issue: #5757
- [x] Ran `yarn fastpass`
- [n/a] Added/updated relevant unit test(s) (and ran `yarn test`)
- [n/a] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/DetailsView/details-view-body.scss [hunk 1]
```diff
@@ -54,7 +54,12 @@
 
         :global(.ms-MessageBar-icon) i {
             margin-left: 12px;
-            color: $secondary-text;
+            color: $always-black;
```
Comment (dbjorge) [inline src/DetailsView/details-view-body.scss:57]: This is an improvement with Accessibility Insights' in-app high contrast setting, but unfortunately it introduces a new issue with native OS forced color settings. For example, here is what I see with Windows 11's "Aquatic" contrast theme:

screenshot of message bar with black icon on dark yellow background

I'd recommend something like this to resolve it for all cases:

```suggestion
            color: $always-black;
            @include ms-high-contrast-override {
                color: inherit;
            }
```

**Comment by dbjorge** [comment]
The failing end to end tests are legitimate, but good - they are failing because this has fixed a previously-grandfathered-in accessibility error caught by two test cases. I don't want to make you ramp up on our e2e test infrastructure just to make a CSS fix, so I'll go ahead and push an update to those tests to your branch here to stop them from expecting the now-fixed issue.

**Review by dbjorge — CHANGES_REQUESTED** [review]
Thanks for the contribution, @patrickhlauke ! Left one suggestion on the `color` attribute, besides that I think we're good to merge.

Someday it would be nice for us to move the message bar styling out of these global details-view-wide styles and into a component stylesheet, but I don't think we need to block this PR's fixes on that, this is a good improvement as-is.

File: src/DetailsView/details-view-body.scss (hunk 1 above)
Comment (patrickhlauke) [reply src/DetailsView/details-view-body.scss:57]: ah, good point. didn't test with actual windows high contrast (on Win 10) itself

**Comment by patrickhlauke** [comment]
thanks @dbjorge - merged your suggestion

**Review by dbjorge — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**